### PR TITLE
For Wayland, run testbed on Ubuntu 24.04

### PR DESCRIPTION
**Pull request by rmartin16**
## Changes
- Closes https://github.com/beeware/toga/issues/2871
- When I use a newer version of mutter locally, the issue goes away...
- Seven [successful runs](https://github.com/beeware/toga/actions/runs/11245617170/job/31267033524) with Ubuntu 24.04

## PR Checklist:

- [x] All new features have been tested
- [x] All new features have been documented
- [X] I have read the **CONTRIBUTING.md** file
- [X] I will abide by the code of conduct

### CODE REVIEW

**Comment by rmartin16** [comment]
well, this is one option; it does require building a wheel for Pillow each time...but it works 😅

**Comment by freakboy3742** [comment]
> well, this is one option; it does require building a wheel for Pillow each time...but it works 😅

Ah- that's because we've pinned such an old version of Pillow. Upgrading the pinned Pillow version is on the todo list; Android wheels are the sticking point. In the meantime, a pillow build is annoying, but isn't *that* onerous - and it's a lot less annoying that having 50% of Toga CI runs fail.

The other annoying inconsistency is that this means the GTK tests are running on 3.12, rather than 3.10 as they are for every other platform. My inclination is to roll back the linux-x11 change to stay on 22.04 for that reason - that also lets us avoid the Pillow build until we can bump the pinned version.

It looks like there was a new version of mutter published in September, so the emergence of this problem looks like it could be consistent with that version update.

**Comment by rmartin16** [comment]
Sounds good; I reverted the changes for `linux-x11` testbed to continue using Ubuntu 22.04.

**Review by freakboy3742 — APPROVED** [review]
I'm going to call this a win. Congrats on a nice bit of debugging.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -232,14 +232,15 @@ jobs:
...
           # The package list should be the same as in tutorial-0.rst, and the BeeWare
-          # tutorial, plus mutter to provide a window manager, and libjpeg-dev for Pillow.
+          # tutorial, plus mutter to provide a window manager.
           pre-command: |
             sudo apt update -y
             sudo apt install -y --no-install-recommends \
               mutter pkg-config python3-dev libgirepository1.0-dev libcairo2-dev \
-              gir1.2-webkit2-4.1 gir1.2-xapp-1.0
+              gir1.2-webkit2-4.1 gir1.2-xapp-1.0 \
+              libjpeg-dev  # remove once the version of Pillow has a wheel on PyPI for Python 3.12
```
Comment (sarayourfriend) [inline .github/workflows/ci.yml:243]: @rmartin16 I came across this line just now and was wondering if the dependency can be removed now? At the time of this PR, Pillow was pinned to 9.2.0 in the testbed, but it's pinned to 11.00.00 now, which has 3.12 wheels on PyPI.

Comment (freakboy3742) [reply .github/workflows/ci.yml:243]: Yes - we should be able to remove this one now.

**Pull request merged**